### DISCUSSION TITLE
gh-pages -- Trying to have my website load all the images properly

### DISCUSSION THREAD

**Discussion by Grichnak** [community/community#70443] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

General

### Body

Hello, 

I am trying to make a portfolio website. I'm using React. I have had a lot of problems with publishing my website on github-pages, but I've mostly made it work (at least it's not a blank page anymore !). Right now my problem is that some icons and images load, but others don't. 

On localhost (how it's supposed to look): 
![image](https://github.com/community/community/assets/112953855/dbb8d1a2-3379-4b97-a3b3-efed735d8cfc)

On gh-pages:
![image](https://github.com/community/community/assets/112953855/118335e9-4e14-4fc0-8a04-491c082a9e54)

The errors I'm getting:
![image](https://github.com/community/community/assets/112953855/37db8d08-f533-474a-96c3-cecc994e1c13)

This is the link to the repository: https://github.com/Grichnak/Portfolio
Since I'm using the gh-pages branch for deploying the website, it's the one that's most up to date. I honsetly have tried to find a solutoin to my problems, but I am still new to a lot of this. If you could even just point me in the right direction I'd be very grateful !

**Comment by ebubekirgungor** [comment]
Hello, you should move all your asset files to public/static/media folder under your src/assets folder.

**Reply by Grichnak** [reply]
Hello,

I can't seem to add files manually to my static folder. What's even more puzzling to me, is that the files that are in my media folder are those that have the "no-display" problem :/ 

These are the files that are in my build/static/media folder right now:
![image](https://github.com/community/community/assets/112953855/0861fcda-6b8e-45e6-b635-1875b95ce293)

These are the images in my src/assets folder:
![image](https://github.com/community/community/assets/112953855/77e777db-93e5-4261-b47f-6ba1a794c696)

**Reply by ebubekirgungor** [reply]
Hello, I found the problem is in https://github.com/Grichnak/Portfolio/blob/gh-pages/asset-manifest.json
You can change directory of images like: 
"static/media/portfolio-nina2.png": "/Portfolio/build/static/media/portfolio-nina2.887dc1881c218e765959.png"

**Reply by ebubekirgungor** [reply]
Also change the `<link rel="icon" href="../assets/logo.png" />` in index file to `<link rel="icon" href="/Portfolio/src/assets/logo.png" />`

**Reply by Grichnak** [reply]
I was looking at the asset-manifest.json when I saw your message XD 

I did try to change the directory but it doesn't seem to change anything unfortunately...

I think I must have done a dumb error at some point, and now my best chance is probably to just create a new clean project and then add my code

**Reply by Grichnak** [reply]
Thank you for trying to help :)

**Reply by ebubekirgungor** [reply]
You are welcome :) Yes creating new project is best solution

**Comment by seyLu** [comment, marked as answer]
#### Hi @Grichnak,

I've [deployed a copy of your site](https://seylu.github.io/Grichnak-Portfolio/). Things to note:

- In your default branch, in this case `master`, update your deploy script to:
```diff
"scripts": {
   "predeploy": "npm run build",
-  "deploy": "gh-pages -b master -d build",
+  "deploy": "gh-pages -d build",
   ...
}
```
- Then on your terminal, run:
```bash
npm run deploy
``` 

- This will create a new build directory which will then be used by the npm package `gh-pages` to deploy what was in the build directory to the `gh-pages` branch.

- Also make sure that on your github settings, it is set to build and deploy from `gh-pages` branch.
![image](https://github.com/community/community/assets/98249191/50aa947e-5f31-4278-93ca-8df18294b2ff)

- After updating the settings, and github doesn't trigger the build and deploy workflow, just run:
```bash
npm run deploy
```
- Which will register a change on the `gh-pages` branch and will trigger the workflow to run.

You can read more on:
- [NPM: gh-pages](https://www.npmjs.com/package/gh-pages#command-line-utility)

**Reply by Grichnak** [reply]
Jesus H. Christ, my brother, I could kiss you right now !

Just as to be sure I understand everything and can help myself the next time I'm in this situation:

-I make some changes to my files,
-I do the whole git add commit push thing
-I do npm run deploy (twice ? )

**Reply by seyLu** [reply]
You don't need to necessarily run `npm run deploy` twice. It's just that when you just changed the repo build and deploy settings but nothing is getting built and deployed -- this is when you run `npm run deploy` to trigger a change in the `gh-pages` branch and have the workflow run.

**Reply by Grichnak** [reply]
Got it ! Thank you very much :)
